packages/template-retail-react-app/app/components/experience/image-with-text/index.jsx

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright (c) 2023, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import React, {useEffect, useState} from 'react'
+import PropTypes from 'prop-types'
+import {Image, Box, Text, Link} from '@chakra-ui/react'
+import DOMPurify from 'dompurify'
```

Comment:
We are currently using a library called [serialize-javascript](https://www.npmjs.com/package/serialize-javascript) in the SDK when serializing the global variables into the returned html. 

Do you think it's possible to use this library to achieve what we want here? Look at the section called **Automatic Escaping of HTML Characters**, you might be able to do something similar, and maybe this works on both server and client, and doesn't increase the bundle size. (Not sure if it will increase the client bundle since we are only using it in the server code, worth a shot tho)

Reply:
It looks like [serialize-javascript](https://www.npmjs.com/package/serialize-javascript) doesn't do what we want as it serializes all the HTML tags. 

Input:
```
<p><em>Text</em> <strong>Overlay</strong>
```


Output:
```
\u003Cp\u003E\u003Cem\u003EText\u003C\u002Fem\u003E \u003Cstrong\u003EOverlay\u003C\u002Fstrong\u003E
```

Reply:
Ok. thanks for looking into that. I'm still a little hesitant on adding that additional library for this situation. In the past we have used a simple utility like [this](https://github.com/SalesforceCommerceCloud/pwa-kit/blob/effd675712279f86cc0586d33ac613ecc49152f0/packages/pwa-kit-runtime/src/utils/ssr-server/utils.js#L76) to sanitize things. I'm sure it could be altered to only escape script tags. (Remember that the data coming back to out knowledge is already sanitized, we'd have to clarify that tho, but from it's current implementation it is)

Reply:
Initially, I started using a RegEx to filter the HTML tags like in this commit https://github.com/SalesforceCommerceCloud/pwa-kit/pull/991/commits/f81d81235d2e05ed3445dbd97111ddbf3e1f43c0
But soon I realized the sanitation problem is tricky and more complex than just removing the script tag.
Example attributes like `onerror` can be added to an img tag with js code or include js code in a `href` attribute.